docs/en/ingest-management/fleet-agent-proxy-support.asciidoc

```diff
@@ -0,0 +1,259 @@
+
+Keep in mind that proxy environment settings may affect other components on the
+host that send HTTP requests.
+
+[discrete]
+[[es-output-proxy-settings]]
+=== Proxy settings in the {agent} policy
+
+Configure proxy settings in the {agent} policy to override proxy settings
+specified by environment variables. For example, you can configure proxy
+settings in the {es} output to override host environment settings.
+
+//QUESTION: Is there a way to set the proxy URL for an input?
+
+The following proxy settings are valid in the {agent} policy:
+
+//QUESTION: Where are these proxy settings valid for Elastic Agent? Just under
+//output? What about input (for services behind a proxy)? I'm not sure how you
+//would configure that if you're not using standalone.
+
+|===
+|Setting | Description
+
+|`proxy_url`
+| (string) URL of the proxy server. If set, the configured URL is used as a
+proxy for all connection attempts by the component. The value may be either a
+complete URL or a `host[:port]`, in which case the `http` scheme is assumed. If
+a value is not specified through the configuration, then proxy environment
+variables are used. The URL accepts optional `username` and `password` settings
+for authenticating with the proxy. For example:
+`http://<username>:<password>@<proxy host>/`.
+
+|`proxy_headers`
+| (string) Additional headers to send to the proxy during CONNECT requests. You
+can use this setting to pass keys/tokens required for authenticating with the
+proxy.
+
+|`proxy_disable`
+| (boolean) If set to `true`, all proxy settings, including the `HTTP_PROXY` and
+`HTTPS_PROXY` environment variables, are ignored.
+
+|===
+
+For {fleet}-managed agents, specify proxy settings in the {kib} UI:
+
+. Log in to {kib} and go to *Management > {fleet}*.
+
+. Click *{fleet} settings*.
+
+. Under *Elasticsearch output configuration (YAML)*, specify proxy settings for
+connecting to {es}. The proxy settings you specify here are applied to all
+{agent}s enrolled in {fleet}.
++
+ADD SCREEN CAPTURE. NEED A REALISTIC EXAMPLE TO PROVIDE HERE.
+
+For standalone agents, specify proxy settings the `elastic-agent.yml` file. For
+example:
+
+[source,yaml]
+----
+outputs:
+  default:
+    type: elasticsearch
+    hosts:
+      - 'http://localhost:9200'
+    proxy_url: 'https://192.0.2.0:8080'
+----
+
+//QUESTION: What should this config actually look like ^^.
+
+For more information, refer to <<elastic-agent-configuration>>.
+
+[discrete]
+[[cli-proxy-settings]]
+=== Command-line flags when enrolling in {fleet}
+
+//QUESTION: It sounds like the user need to visit Fleet to load policies before
+//running the command? Or can we just assume users will do that first before
+//getting to this step?
+
+Specify proxy settings on the command line when you are enrolling in {fleet} and
+the proxy configuration is not yet available. During enrollment, the proxy
+settings specified at the command line are copied to internal configuration
+files.
```

Comment:
Are the proxy settings copied to internal files? I think in my testing I still had to specify a proxy if one was between the agent and ES

Reply:
I saw that in one of the GitHub issues, but it's possible that the info was old. It's hard to tell sometimes what is the actual implementation vs just discussion. :-/

Reply:
The `--proxy-url` value used when installing/enrolling in fleet is added as `fleet.proxy_url`, but it's not automatically  added to the `outputs` section (or `agent.download`).
Outputs settings can be added through the UI (in the settings panel).
However there's no easy way to set the `agent.download.proxy_url` so must be added manually to `fleet.yml`.